pynbody/units.py

```diff
@@ -280,6 +280,8 @@ class NoUnit(UnitBase):
 
     def __init__(self):
         self._no_unit = True
+        self._bases = []
+        self._powers = []
```

Comment:
This again suggests to me a better way forward is a `__hash__` so that we don't need to worry about exposing internal state outside the classes themselves